adapters/jixie/jixie.go

```diff
@@ -0,0 +1,161 @@
+package jixie
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type JixieAdapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Jixie adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	bidder := &JixieAdapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+// Adding header fields to request header
+func addHeaderIfNonEmpty(headers http.Header, headerName string, headerValue string) {
+	if len(headerValue) > 0 {
+		headers.Add(headerName, headerValue)
+	}
+}
+
+//func buildEndpoint(endpoint string, timeout int64) string {
+//	return endpoint + "?pstimeout=" + strconv.FormatInt(timeout, 10)
+//}
+
+func (a *JixieAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("No Imps in Bid Request"),
+		}}
+	}
+
+	for _, imp := range request.Imp {
+
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: fmt.Sprintf("Impression id=%s has an Error: %s", imp.ID, err.Error()),
+			})
+			continue
+		}
+		var jxExt openrtb_ext.ExtImpJixie
+		if err := json.Unmarshal(bidderExt.Bidder, &jxExt); err != nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: fmt.Sprintf("Impression id=%s, has invalid Ext", imp.ID),
+			})
+			continue
+		}
+	}
+
+	data, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	if request.Device != nil {
+		addHeaderIfNonEmpty(headers, "User-Agent", request.Device.UA)
+		addHeaderIfNonEmpty(headers, "X-Forwarded-For", request.Device.IP)
+		addHeaderIfNonEmpty(headers, "Accept-Language", request.Device.Language)
+		if request.Device.DNT != nil {
+			addHeaderIfNonEmpty(headers, "DNT", strconv.Itoa(int(*request.Device.DNT)))
+		}
+	}
```

Comment:
Consider a test that sets the `"Accept-Language"` and `"DNT"` the headers: 
```
72     "httpCalls": [{
73       "expectedRequest": {
74         "uri": "https://hb.jixie.io/v2/hbsvrpost?hbofftest=1",
75         "headers": {
76           "Accept": [
77             "application/json"
78           ],
   +         "Accept-Language": [
   +            "anyAcceptLanguageValue"
   +          ],
   +         "DNT": [
   +            "anyDntValue"
   +          ],
79           "Content-Type": [
80             "application/json;charset=utf-8"
81           ],
82           "X-Forwarded-For": [
83             "111.222.333.444"
84           ],
85           "Referer": [
86             "http://www.publisher.com/today/site?param1=a&param2=b"
87           ],
88           "User-Agent": [
89             "Mozilla/5.0 (Linux; Android 8.0.0; SM-G960F Build/R16NW) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/62.0.3202.84 Mobile Safari/537.36"
90           ]
91         },
92         "body": {
adapters/jixie/jixietest/exemplary/banner-and-video-site.json
```

Reply:
decided not to add the lang and dnt to the headers after all. Changed jixie.go 0xaa2d1d5